pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/ControllerBrokerSelector.java

```
@@ -0,0 +1,122 @@
+    private static final Random RANDOM = new Random();
+    private List<String> _brokers;
+
+    private static class BrokerInstance {
+        @JsonProperty("host")
+        public String _host;
+
+        @JsonProperty("port")
+        public int _port;
+
+        @JsonProperty("instanceName")
+        public String _instanceName;
+    }
+
+    private class BrokerUpdater extends Thread {
+        private final AsyncHttpClient _client;
+        private static final String ADDRESS_FORMAT = "http://%s:%d/v2/brokers/tables";
+        private final ObjectMapper _objectMapper = new ObjectMapper();
+        private final String _address;
+        private final TypeReference<Map<String, List<BrokerInstance>>> _responseTypeRef =
+            new TypeReference<Map<String, List<BrokerInstance>>>() { };
+
+
+        public BrokerUpdater(String controllerHost, int controllerPort) {
+            _client = Dsl.asyncHttpClient();
+            _address = String.format(ADDRESS_FORMAT, controllerHost, controllerPort);
+        }
+
+        public void updateBrokers() {
+            BoundRequestBuilder getRequest = _client.prepareGet(_address);
+            Future<Response> responseFuture = getRequest.addHeader("accept", "application/json").execute();
+            try {
+                Response response = responseFuture.get();
+                String responseBody = response.getResponseBody(StandardCharsets.UTF_8);
+                Map<String, List<BrokerInstance>> responses = _objectMapper.readValue(responseBody, _responseTypeRef);
+                List<String> brokers = new ArrayList<>();
+                for (Map.Entry<String, List<BrokerInstance>> b: responses.entrySet()) {
+                    b.getValue().forEach(br -> {
+                        brokers.add(br._host + ":" + br._port);
+                    });
+                }
+                _brokers = brokers;
+            } catch (Exception e) {
+                e.printStackTrace();
```

Are we swallowing exception here?

---

Ack.
Intellij defaults :)
Added logging wherever needed.